pkg/ssh/shell/setupBuildEnv.sh

```diff
@@ -1,29 +1,8 @@
 #!/usr/bin/env bash
 set -e
 #name:TASK [install gcc if not available]
-sudo apt-get -y -o DPkg::Lock::Timeout=120 update
-while ! gcc --version >/dev/null 2>&1; do
-    echo "GCC is not installed. Trying to install..."
-    DEBIAN_FRONTEND=noninteractive sudo apt-get -y -o DPkg::Lock::Timeout=120 install gcc
-    if [ $? -ne 0 ]; then
-        echo "Failed to install GCC. Retrying in 10 seconds..."
-        sleep 10
-    fi
-done
+export DEBIAN_FRONTEND=noninteractive
+until sudo apt-get -y update -o DPkg::Lock::Timeout=120; do sleep 10 && echo "Try again"; done
+until sudo apt-get -y install -o DPkg::Lock::Timeout=120 gcc; do sleep 10 && echo "Try again"; done
 #name:TASK [install go]
-install_go() {
-  local GOFILE
-  if [[ "$(uname -m)" == "aarch64" ]]; then
-    GOFILE="go{{ .GoVersion }}.linux-arm64.tar.gz"
-  else
-    GOFILE="go{{ .GoVersion }}.linux-amd64.tar.gz"
-  fi
-  cd ~
-  sudo rm -rf $GOFILE go
-  wget -q -nv https://go.dev/dl/$GOFILE
-  tar xfz $GOFILE
-  echo >> ~/.bashrc
-  echo export PATH=\$PATH:~/go/bin:~/bin >> ~/.bashrc
-  echo export CGO_ENABLED=1 >> ~/.bashrc
-}
-go version || install_go
+go version || sudo snap install go --classic
```

Comment:
1. we are loosing control of golang version here (.GoVersion). Do we want that? we probably can use --channel=... here
2. It is posible for snap latest version to become obsolete regarding vms/avago compilation needs? in this case a tar give best control.
3. Besides that, probably it is a bit more custom to use apt instead of snap.